packages/wallets/particle/src/adapter.ts

```diff
@@ -0,0 +1,179 @@
+
+    private _connecting: boolean;
+    private _wallet: SolanaWallet | null;
+    private _publicKey: PublicKey | null;
+    private _config: ParticleAdapterConfig;
+    private _readyState: WalletReadyState =
+        typeof window === 'undefined' ? WalletReadyState.Unsupported : WalletReadyState.Loadable;
+
+    constructor(config: ParticleAdapterConfig) {
+        super();
+        this._connecting = false;
+        this._publicKey = null;
+        this._wallet = null;
+        this._config = config;
+    }
+
+    get publicKey(): PublicKey | null {
+        return this._publicKey;
+    }
+
+    get connecting(): boolean {
+        return this._connecting;
+    }
+
+    get connected(): boolean {
+        return !!this._publicKey;
+    }
+
+    get readyState(): WalletReadyState {
+        return this._readyState;
+    }
+
+    async connect(): Promise<void> {
+        try {
+            if (this.connected || this.connecting) return;
+            if (this._readyState !== WalletReadyState.Loadable) throw new WalletNotReadyError();
+
+            this._connecting = true;
+
+            let ParticleClass: typeof ParticleNetwork;
+            try {
+                ({ default: ParticleClass } = await import('@particle-network/auth'));
+            } catch (error: any) {
+                throw new WalletLoadError(error?.message, error);
+            }
+
+            let wallet: SolanaWallet;
+            try {
+                wallet = window.particle || new SolanaWallet(new ParticleClass(this._config).auth);
+            } catch (error: any) {
+                throw new WalletConfigError(error?.message, error);
+            }
+
+            try {
+                await wallet.connect();
+            } catch (error: any) {
+                throw new WalletConnectionError(error?.message, error);
+            }
+
+            if (!wallet.publicKey()) throw new WalletAccountError();
+
+            let publicKey: PublicKey;
+            try {
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                publicKey = wallet.publicKey()!;
```

Comment:
Why would this throw an error, or be undefined? Also, we should copy the public key bytes and instantiate a `PublicKey` here.